### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by Manpreet1502**

### CODE REVIEW

File: core_java/.idea/.gitignore [hunk 1]
```diff
@@ -0,0 +1,2 @@
+# Default ignored files
```
Comment (jarviscanada) [inline core_java/.idea/.gitignore]: remove .idea

File: core_java/grep/src/main/java/ca/jrvs/apps/grep/JavaGrep.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+public interface JavaGrep {
+
+    void process() throws IOException;
+
+    /*traverse given directory and returns all files*/
```
Comment (jarviscanada) [inline core_java/grep/src/main/java/ca/jrvs/apps/grep/JavaGrep.java]: remove these comments

File: core_java/grep/src/main/java/ca/jrvs/apps/grep/JavaGrepImp.java [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+        List<File> files = listFiles(rootPath);
+        List<String> matchedLines =  new ArrayList<>();;
+
+        for(File file : files){
+            List<String> lines = readLines(file);
+            boolean matchFound = false;
+            for(String line: lines){
+                matchFound = containsPattern(line);
+
+                //check if match is found
```
Comment (jarviscanada) [inline core_java/grep/src/main/java/ca/jrvs/apps/grep/JavaGrepImp.java]: remove

File: core_java/grep/src/main/java/ca/jrvs/apps/grep/JavaGrepImp.java [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+        if(matchedLines!=null){
+            this.writeToFile(matchedLines);
+            System.out.println("Patterns has been written to File");
+        }else{
+            System.out.println("Pattern Not found");
+        }
+
+    }
+
+    //traverse a directory and return all the files
```
Comment (jarviscanada) [inline core_java/grep/src/main/java/ca/jrvs/apps/grep/JavaGrepImp.java]: not javadoc

File: core_java/grep/src/main/java/ca/jrvs/apps/grep/JavaGrepImp.java [hunk 5]
```diff
@@ -0,0 +1,199 @@
...
+
+        // Check if rootDir is Directory
+        if(!directory.exists() || !directory.isDirectory()){
+            throw new IllegalArgumentException(String.format("%s is not a directory", rootDir));
+        }
+
+        List<File> fileList = new ArrayList<>();
+        File rootdir = new File(rootDir);
+
+        //Get the name of directories contained in the dir
```
Comment (jarviscanada) [inline core_java/grep/src/main/java/ca/jrvs/apps/grep/JavaGrepImp.java]: keep this

File: core_java/grep/src/main/java/ca/jrvs/apps/grep/JavaGrepImp.java [hunk 6]
```diff
@@ -0,0 +1,199 @@
...
+
+    }
+
+    //traverse a directory and return all the files
+    @Override
+    public List<File> listFiles(String rootDir)
+    {
+        File directory = new File(rootDir);
+
+        // Check if rootDir is Directory
```
Comment (jarviscanada) [inline core_java/grep/src/main/java/ca/jrvs/apps/grep/JavaGrepImp.java]: not necessary

File: core_java/twitter/src/main/java/ca/jrvs/apps/twitter/dao/helper/TwitterHttpHelper.java [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+    private OAuthConsumer oAuthConsumer;
+
+    public TwitterHttpHelper() {
+    }
+
+    @Autowired
+    public TwitterHttpHelper(@Value("${consumerKey}") String consumerKey,
+                             @Value("${consumerSecret}") String consumerSecret,
+                             @Value("${accessToken}") String accessToken,
+                             @Value("${accessTokenSecret}") String accessTokenSecret) {
```
Comment (jarviscanada) [inline core_java/twitter/src/main/java/ca/jrvs/apps/twitter/dao/helper/TwitterHttpHelper.java]: good joob

File: core_java/twitter/src/main/java/ca/jrvs/apps/twitter/service/TwitterService.java [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+import ca.jrvs.apps.twitter.dao.CrdDao;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.util.ArrayList;
+import java.util.List;
+
+@org.springframework.stereotype.Service
```
Comment (jarviscanada) [inline core_java/twitter/src/main/java/ca/jrvs/apps/twitter/service/TwitterService.java]: remove line 14

File: core_java/twitter/src/main/java/ca/jrvs/apps/twitter/dao/helper/TwitterHttpHelper.java (hunk 7 above)
Comment (Manpreet1502) [reply core_java/twitter/src/main/java/ca/jrvs/apps/twitter/dao/helper/TwitterHttpHelper.java]: Thanks

**Pull request merged**